### PR TITLE
TestStatusLocalLogs - add retry and report full errors

### PR DESCRIPTION
**Pull request by tamird**
@BramGruneir for you :heart_eyes_cat:

### CODE REVIEW

File: server/status_test.go [hunk 1]
```diff
@@ -242,34 +249,44 @@ func getRequest(t *testing.T, ts TestServer, path string) []byte {
...
-	return body
+	t.Fatalf("There was an error retrieving %s", path)
+	return []byte("")
 }
 
 // startServer will start a server with a short scan interval, wait for
 // the scan to complete, and return the server. The caller is
 // responsible for stopping the server.
-// TODO(Bram): Add more nodes.
+// TODO(bram): Add more nodes.
```
Comment (BramGruneir) [inline server/status_test.go]: TODOs need to have only lower case names?

File: server/status_test.go [hunk 2]
```diff
@@ -242,34 +249,44 @@ func getRequest(t *testing.T, ts TestServer, path string) []byte {
...
+		if resp.StatusCode != http.StatusOK {
+			t.Logf("unexpected status code: %v", resp.StatusCode)
+			continue
+		}
+		returnedContentType := resp.Header.Get(util.ContentTypeHeader)
+		if returnedContentType != util.JSONContentType {
+			t.Logf("unexpected content type: %v", returnedContentType)
+			continue
+		}
+		body, err := ioutil.ReadAll(resp.Body)
```
Comment (BramGruneir) [inline server/status_test.go]: Can you move this up? The body may contain the server error. It should get the body before checking the status.

**Comment by BramGruneir** [comment]
What prompted this?  Another test failure?

File: server/status_test.go (hunk 1 above)
Comment (tamird) [inline server/status_test.go]: no, but there were more instances of `TODO(bram)` than there were of `TODO(Bram)`

File: server/status_test.go (hunk 2 above)
Comment (tamird) [inline server/status_test.go]: and then log the body in case of error? I don't totally understand what you're suggesting

**Comment by tamird** [comment]
> What prompted this? Another test failure?

See the commit message, this is #1940 which I just observed again in #1839.

File: server/status_test.go (hunk 2 above)
Comment (BramGruneir) [inline server/status_test.go]: yes, exactly, add the body to the logs for the other failures.

**Comment by tamird** [comment]
@BramGruneir PTAL

**Comment by BramGruneir** [comment]
LGTM

**Pull request merged**